### PR TITLE
Eliminate conversion loops use assembly

### PR DESCRIPTION
**Pull request by i-stam**

### CODE REVIEW

File: contracts/externals/SafeMath.sol [hunk 1]
```diff
@@ -23,68 +23,68 @@
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-pragma solidity ^0.4.24;
+ pragma solidity ^0.5.0;
```
[the remark this answers is not in the document]
Comment (i-stam) [reply contracts/externals/SafeMath.sol]: yes, usually the 0.5.0 ones are external (using pragmas for generic use), we are trying to use the latest solc release

File: contracts/wallet.sol [hunk 2]
```diff
@@ -704,79 +704,58 @@ contract Wallet is ENSResolvable, Vault, GasTopUpLimit, LoadLimit {
...
+        success := call(gas, _to, _value, add(_data, 0x20), mload(_data), 0, 0)
     }
+}
+
 
     /// @dev This function converts to an address
     /// @param _bts bytes
     /// @param _from start position
-    function _bytesToAddress(bytes _bts, uint _from) private pure returns (address) {
+    function _bytesToAddress(bytes memory _bts, uint _from) public pure returns (address payable) {
```
Comment (draganm) [inline contracts/wallet.sol]: probably should not be public.

**Comment by mischat** [comment]
I don't mind this being merged, but this seems to touch a lot of things, not just the assembly code for the bytes conversion loops in the wallet. 

Aside from this and need to rebase, what do we think about the test coverage of these two conversions loops, are we happy with them ?

**Comment by mischat** [comment]
you also mentioned that your assembly code was audited by trail of bits, can you share this please? or was it in the chat ? @i-stam

**Comment by i-stam** [comment]
Yes, in the slack channel i guess, do we have a copy? I don't think so...never mind I ll create a new PR and move them possibly in a separate file...

**Pull request closed without merging**